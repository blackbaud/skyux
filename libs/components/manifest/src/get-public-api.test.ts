describe('get-public-api', () => {
  function setup(options: { publicApi: Record<string, unknown> }): void {
    jest.mock('../public-api.json', () => options.publicApi);
  }

  afterEach(() => {
    jest.resetAllMocks();
    jest.resetModules();
  });

  it('should return the public API', async () => {
    const publicApi = {
      packages: {
        '@company/components': {},
      },
    };

    setup({ publicApi });

    const { getPublicApi } = await import('./get-public-api');

    const result = getPublicApi();

    expect(result).toEqual(publicApi);
  });

  it('should return the public API for a specific docs ID', async () => {
    const publicApi = {
      packages: {
        '@company/components': [
          {
            docsId: 'foo',
            isInternal: false,
          },
          {
            docsId: 'bar',
            isInternal: false,
            docsIncludeIds: ['foo', 'FooTesting', 'FooTestingInternal'],
          },
          {
            docsId: 'ignored',
            isInternal: false,
          },
        ],
        '@company/components/testing': [
          {
            docsId: 'FooTesting',
            isInternal: false,
          },
          {
            docsId: 'FooTestingInternal',
            isInternal: true,
          },
        ],
      },
    };

    setup({ publicApi });

    const { getPublicApiByDocsId } = await import('./get-public-api');

    const result = getPublicApiByDocsId('bar');

    expect(result).toMatchSnapshot();
  });
});
