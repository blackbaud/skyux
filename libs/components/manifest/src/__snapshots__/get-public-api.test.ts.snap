// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-public-api should return the public API for a specific docs ID 1`] = `
Object {
  "packages": Object {
    "@company/components": Array [
      Object {
        "docsId": "bar",
        "docsIncludeIds": Array [
          "foo",
          "FooTesting",
          "FooTestingInternal",
        ],
        "isInternal": false,
      },
      Object {
        "docsId": "foo",
        "isInternal": false,
      },
    ],
    "@company/components/testing": Array [
      Object {
        "docsId": "FooTesting",
        "isInternal": false,
      },
    ],
  },
}
`;
